components/TimeSeries.vue

```diff
@@ -0,0 +1,353 @@
+<template>
+  <!-- Per each time series, use one accordion component with one item, so we can easily initialise them all as open with active-item-key -->
+  <CAccordion
+    :style="accordionStyle"
+    :active-item-key="isOpen ? props.seriesId : undefined"
+  >
+    <CAccordionItem :item-key="seriesId" class="border-0">
+      <CAccordionHeader class="border-top" @click="handleAccordionToggle">
+        {{ seriesMetadata?.label }}
+      </CAccordionHeader>
+      <CAccordionBody>
+        <div
+          :id="containerId"
+          :class="`chart-container ${props.hideTooltips ? hideTooltipsClassName : ''}`"
+          :style="{ zIndex, height: 'fit-content' }"
+          @mousemove="onMove"
+          @touchmove="onMove"
+          @touchstart="onMove"
+        />
+      </CAccordionBody>
+    </CAccordionItem>
+  </CAccordion>
+</template>
+
+<script lang="ts" setup>
+import throttle from "lodash.throttle";
+import * as Highcharts from "highcharts";
+import accessibilityInitialize from "highcharts/modules/accessibility";
+import exportingInitialize from "highcharts/modules/exporting";
+import exportDataInitialize from "highcharts/modules/export-data";
+import offlineExportingInitialize from "highcharts/modules/offline-exporting";
+
+import { highchartsColors, plotBandsColor, plotLinesColor } from "./utils/charts";
+import type { DisplayInfo } from "~/types/apiResponseTypes";
+
+const props = defineProps<{
+  seriesId: string
+  index: number
+  openedAccordions: string[]
+  hideTooltips: boolean
+}>();
+
+const emit = defineEmits(["toggleOpen"]);
+accessibilityInitialize(Highcharts);
+exportingInitialize(Highcharts);
+exportDataInitialize(Highcharts);
+offlineExportingInitialize(Highcharts);
+
+const appStore = useAppStore();
+
+let chart: Highcharts.Chart;
+const chartBackgroundColor = "transparent";
+const chartBackgroundColorOnExporting = "white";
+const hideResetZoomButtonClassName = "hide-reset-zoom-button";
+const hideTooltipsClassName = "hide-tooltips";
+const accordionBodyYPadding = 8;
+const accordionStyle = {
+  "--cui-accordion-btn-focus-box-shadow": "none",
+  "--cui-accordion-bg": "rgba(255, 255, 255, 0.7)",
+};
+// We need each chart to have a higher z-index than the next one so that the exporting context menu is always on top and clickable.
+// Also, they should be at least 3 so that they are above .accordion-button:focus
+const zIndex = (Object.keys(appStore.timeSeriesData!).length - props.index) + 3;
+const yUnits = props.seriesId === "dead" ? "deaths" : "cases"; // TODO: Make this depend on a 'units' property in metadata. https://mrc-ide.myjetbrains.com/youtrack/issue/JIDEA-117/
+const minAccordionHeight = 150;
+const minTotalAccordionHeight = 500;
+
+const isOpen = computed(() => props.openedAccordions.includes(props.seriesId));
```

Comment:
This *was* a bit of efficiency because we were already passing openedAccordions in order to use openedAccordions.length. But now all that is getting handled in the intermediate component TimeSeriesList.